lhotse/bin/modes/cut.py

```diff
@@ -0,0 +1,114 @@
+from pathlib import Path
+from typing import Tuple
+
+import click
+import numpy as np
+
+from lhotse.bin.modes.cli_base import cli
+from lhotse.cut import make_cuts_from_supervisions, CutSet, mix_stereo_cut_set
+from lhotse.features import FeatureSet
+from lhotse.manipulation import split
+from lhotse.supervision import SupervisionSet
+from lhotse.utils import Pathlike, fix_random_seed
+
+__all__ = ['cut', 'simple', 'random_overlayed', 'stereo_overlayed']
+
+
+@cli.group()
+def cut():
+    """Group of commands used to create CutSets."""
+    pass
+
+
+@cut.command()
+@click.argument('supervision_manifest', type=click.Path(exists=True, dir_okay=False))
+@click.argument('feature_manifest', type=click.Path(exists=True, dir_okay=False))
+@click.argument('output_cut_manifest', type=click.Path())
+def simple(
+        supervision_manifest: Pathlike,
+        feature_manifest: Pathlike,
+        output_cut_manifest: Pathlike,
+):
+    """
+    Create a CutSet stored in OUTPUT_CUT_MANIFEST that contains supervision regions from SUPERVISION_MANIFEST
+    and features supplied by FEATURE_MANIFEST. This is the simplest way to create Cuts.
+    """
+    supervision_set = SupervisionSet.from_yaml(supervision_manifest)
+    feature_set = FeatureSet.from_yaml(feature_manifest)
+    cut_set = make_cuts_from_supervisions(supervision_set=supervision_set, feature_set=feature_set)
+    cut_set.to_yaml(output_cut_manifest)
+
+
+@cut.command()
+@click.argument('supervision_manifest', type=click.Path(exists=True, dir_okay=False))
+@click.argument('feature_manifest', type=click.Path(exists=True, dir_okay=False))
+@click.argument('output_cut_manifest', type=click.Path())
+@click.option('-r', '--random-seed', default=42, type=int, help='Random seed value.')
+@click.option('-s', '--snr-range', type=(float, float), default=(20, 20),
+              help='Range of SNR values (in dB) that will be uniformly sampled in order to overlay the signals.')
+@click.option('-o', '--offset-range', type=(float, float), default=(0.5, 0.5),
+              help='Range of relative offset values (0 - 1), which will offset the "right" signal by this many times '
+                   'the duration of the "left" signal. It is uniformly sampled for each overlay operation.')
+def random_overlayed(
+        supervision_manifest: Pathlike,
+        feature_manifest: Pathlike,
+        output_cut_manifest: Pathlike,
+        random_seed: int,
+        snr_range: Tuple[float, float],
+        offset_range: Tuple[float, float]
+):
+    """
+    Create a CutSet stored in OUTPUT_CUT_MANIFEST that contains supervision regions from SUPERVISION_MANIFEST
+    and features supplied by FEATURE_MANIFEST. It first creates a trivial CutSet, splits it into two equal, randomized
+    parts and overlays their features to create a mix.
+    The parameters of the mix are controlled via SNR_RANGE and OFFSET_RANGE.
+    """
```

Comment:
So will the mix's duration always cover both entire original recordings?  Are the random pairs selected without regard to length?
For some separation tasks, ragged edges may make it artificially easier.

Reply:
These're good questions. When building the Cut/CutSet I wanted to have a starting point that performs the mix in *some* way to test it out - so I figured out let's start with random. 

You are correct that in this mode the mix duration always covers both recordings, as the underlying mix/overlay code pads both cuts with low energy frames so that they can be added together. We could add another parameter like `max_length` (or `max_length_range`) to truncate (and that would likely create ragged edges).

The random pairs are indeed selected without regard to length. I didn't want to make any assumptions at this point; I'm happy to modify it whichever way you, @mpariente or @popcornell suggest. We can always add more parameters or modes to create mixes in different ways.

Reply:
In predefined separation datasets (wsj0-2mix, LibriMix), the sources are summed without offset and mainly without considering the respective length of the sources. 
The mixture can be used in the `min` mode where the mixture is truncated to the shortest source or the `max` mode, where the shortest is padded with zeros. 
For training separation, we usually use the `min` mode because more challenging. 

> We can always add more parameters or modes to create mixes in different ways.

I agree with that, this is not critical and it looks fine for now.

Reply:
Thanks, that's great feedback. In the next PR that I'm preparing I will add the option to mix in min/max modes.